llama_index/indices/service_context.py

```diff
@@ -71,14 +81,20 @@ def from_defaults(
         """
         callback_manager = callback_manager or CallbackManager([])
         llm_predictor = llm_predictor or LLMPredictor()
+        llm_predictor._callback_manager = callback_manager
```

Comment:
Do not set private attribute. The point of a private attribute is to communicate that it will not be changed by external classes.

If this attribute setting must happen, create a setter for it.